### PR TITLE
Feat:  notification data models #150

### PR DESCRIPTION
**Pull request by kevin-hashimoto**

### CODE REVIEW

File: backend/lcfs/db/models/NotificationChannelSubscription.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+class NotificationChannelSubscription(BaseModel, Auditable):
+     __tablename__ = 'notification_channel_subscription'
+     __table_args__ = {'comment': "Represents a user's subscription to notification events"}
+
+     id = Column(Integer, primary_key=True, autoincrement=True)
+
+     is_enabled = Column(Boolean, default=False)
+
+     user_id = Column(Integer, ForeignKey('user.id'))
+     channel_id = Column(Integer, ForeignKey('notification_channel.id'))
```
Comment (AlexZorkin) [inline backend/lcfs/db/models/NotificationChannelSubscription.py:14]: these foreign keys will have to be updated, notification_channel.notification_channel_id
a bit redundant but for database standards we'll have to bear it.

File: backend/lcfs/db/models/NotificationMessage.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+from lcfs.db.base import BaseModel, Auditable
+from sqlalchemy import Column, Integer, ForeignKey, Boolean
+from sqlalchemy.orm import relationship
+
+class NotificationMessage(BaseModel, Auditable):
+    __tablename__ = 'notification_message'
+    __table_args__ = {'comment': "Represents a notification message sent to an application user"}
+
+    id = Column(Integer, primary_key=True, autoincrement=True)
```
Comment (AlexZorkin) [inline backend/lcfs/db/models/NotificationMessage.py:9]: notification_message_id

File: backend/lcfs/db/models/NotificationMessage.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    is_read = Column(Boolean, default=False)
+    is_warning = Column(Boolean, default=False)
+    is_error = Column(Boolean, default=False)
+    is_archived = Column(Boolean, default=False)
+
+    origin_user_id = Column(Integer,ForeignKey('user.id'))
+    related_organization_id = Column(Integer, ForeignKey('organization.id'))
```
Comment (AlexZorkin) [inline backend/lcfs/db/models/NotificationMessage.py:17]: foreign key ids to be updated

File: backend/lcfs/db/models/NotificationType.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+     EXCLUSION_REPORT_REJECTED = "Exclusion Report Rejected"
+     EXCLUSION_REPORT_REQUESTED_SUPPLEMENTAL = \
+          "Exclusion Report Requested Supplemental"
+
+
+class Notificationtype(BaseModel, Auditable):
+     __tablename__ = 'notification_type'
+     __table_args__ = {'comment': "Represents a Notification type"}
+
+     id = Column(Integer, primary_key=True, autoincrement=True)
```
Comment (AlexZorkin) [inline backend/lcfs/db/models/NotificationType.py:74]: notification_type_id

**Review by AlexZorkin — CHANGES_REQUESTED** [review]

File: backend/lcfs/db/models/NotificationMessage.py (hunk 3 above)
Comment (kevin-hashimoto) [reply backend/lcfs/db/models/NotificationMessage.py]: to keep consistency, i updated the ID keys with prefixes for the organizationa, user and role models.

**Review by AlexZorkin — APPROVED** [review]

**Pull request merged**